src/NCronJob/Registry/JobAttributeCache.cs

```diff
@@ -0,0 +1,14 @@
+using System.Collections.Concurrent;
+using System.Reflection;
+
+namespace NCronJob;
+internal static class JobAttributeCache
+{
+    private static readonly ConcurrentDictionary<object, JobExecutionAttributes> Cache = new();
+
+    public static JobExecutionAttributes GetJobExecutionAttributes(Type jobType) =>
+        Cache.GetOrAdd(jobType, (type) => JobExecutionAttributes.CreateFromType(type as Type));
```

Comment:
Do we need the safe cast? If we hard-cast to `Type` we can remove the null-check.

Reply:
hard casting works, done